apis/nucleus/src/components/Sheet.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState, useContext } from 'react';
+import useLayout from '../hooks/useLayout';
+import getObject from '../object/get-object';
+import Cell from './Cell';
+import uid from '../object/uid';
+import { resolveBgColor, resolveBgImage } from '../utils/background-props';
+import InstanceContext from '../contexts/InstanceContext';
+
+function getCellRenderer(cell, halo, initialSnOptions, initialSnPlugins, initialError, onMount) {
+  const { x, y, width, height } = cell.bounds;
+  return (
+    <div style={{ left: `${x}%`, top: `${y}%`, width: `${width}%`, height: `${height}%`, position: 'absolute' }}>
+      <Cell
+        ref={cell.cellRef}
+        halo={halo}
+        model={cell.model}
+        currentId={cell.currentId}
+        initialSnOptions={initialSnOptions}
+        initialSnPlugins={initialSnPlugins}
+        initialError={initialError}
+        onMount={onMount}
+      />
+    </div>
+  );
+}
+
+function Sheet({ model, halo, initialSnOptions, initialSnPlugins, initialError, onMount }) {
+  const { root } = halo;
+  const [layout] = useLayout(model);
+  const { theme: themeName } = useContext(InstanceContext);
+  const [cells, setCells] = useState([]);
+  const [bgColor, setBgColor] = useState(undefined);
+  const [bgImage, setBgImage] = useState(undefined);
+
+  /// For each object
+  useEffect(() => {
+    if (layout) {
+      const fetchObjects = async () => {
+        const lCells = layout.cells;
+        const vs = await Promise.all(lCells.map((c) => getObject({ id: c.name }, halo)));
+        const cs = lCells.map((c, i) => {
+          let mounted;
+          const mountedPromise = new Promise((resolve) => {
+            mounted = resolve;
+          });
+          return {
+            model: vs[i].model,
+            id: c.name,
+            bounds: c.bounds,
+            cellRef: React.createRef(),
+            currentId: uid(),
```

Comment:
It feels wrong to create a new currentId each time the sheet layout changes. (but maybe it is OK)

Reply:
yea I agree, need to think about how to handle the lifecycle of the sheet better.